x/group/keeper/invariants_test.go

```diff
@@ -0,0 +1,587 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/libs/log"
+
+	tmtime "github.com/tendermint/tendermint/libs/time"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/cosmos/cosmos-sdk/x/group/internal/orm"
+	"github.com/cosmos/cosmos-sdk/x/group/keeper"
+)
+
+type invariantTestSuite struct {
+	suite.Suite
+
+	ctx       sdk.Context
+	cdc       *codec.ProtoCodec
+	key       *storetypes.KVStoreKey
+	blockTime time.Time
+}
+
+func TestInvariantTestSuite(t *testing.T) {
+	suite.Run(t, new(invariantTestSuite))
+}
+
+func (s *invariantTestSuite) SetupSuite() {
+	interfaceRegistry := types.NewInterfaceRegistry()
+	group.RegisterInterfaces(interfaceRegistry)
+	cdc := codec.NewProtoCodec(interfaceRegistry)
+	key := sdk.NewKVStoreKey(group.ModuleName)
+	db := dbm.NewMemDB()
+	cms := store.NewCommitMultiStore(db)
+	cms.MountStoreWithDB(key, storetypes.StoreTypeIAVL, db)
+	_ = cms.LoadLatestVersion()
+	sdkCtx := sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger())
+
+	s.ctx = sdkCtx
+	s.cdc = cdc
+	s.key = key
+	s.blockTime = tmtime.Now()
+
+}
+
+func (s *invariantTestSuite) TestTallyVotesInvariant() {
+	sdkCtx, _ := s.ctx.CacheContext()
+	curCtx, cdc, key := sdkCtx, s.cdc, s.key
+	prevCtx, _ := curCtx.CacheContext()
+	prevCtx = prevCtx.WithBlockHeight(curCtx.BlockHeight() - 1)
+
+	// Proposal Table
+	proposalTable, err := orm.NewAutoUInt64Table([2]byte{keeper.ProposalTablePrefix}, keeper.ProposalTableSeqPrefix, &group.Proposal{}, cdc)
+	s.Require().NoError(err)
+
+	_, _, addr1 := testdata.KeyTestPubAddr()
+	_, _, addr2 := testdata.KeyTestPubAddr()
+
+	specs := map[string]struct {
+		prevProposal *group.Proposal
+		curProposal  *group.Proposal
+		expBroken    bool
+	}{
+		"invariant not broken": {
+			prevProposal: &group.Proposal{
+				ProposalId:          1,
+				Address:             addr1.String(),
+				Proposers:           []string{addr1.String()},
+				SubmittedAt:         prevCtx.BlockTime(),
+				GroupVersion:        1,
+				GroupAccountVersion: 1,
+				Status:              group.ProposalStatusSubmitted,
+				Result:              group.ProposalResultUnfinalized,
+				VoteState:           group.Tally{YesCount: "1", NoCount: "0", AbstainCount: "0", VetoCount: "0"},
+				Timeout:             s.blockTime.Add(time.Second * 600),
```

Comment:
```suggestion
				Timeout:             prevCtx.BlockTime().Add(time.Second * 600),
```

ditto for all the proposals below
let's just be consistent with the proposal `SubmittedAt` (although it might not have direct impact on the tests themselves)